src/components/Pages/MetadataForm.jsx

```diff
@@ -254,6 +262,42 @@ class MetadataForm extends FormClassTemplate {
     return contactsRef.push(contact).getKey();
   }
 
+  async handleUpdateDraftDOI() {
```

Comment:
is this the same as https://github.com/cioos-siooc/metadata-entry-form/blob/feature/update-doi-on-submit/src/components/Tabs/IdentificationTab.jsx/#L143? Can these be combined to reduce duplicate code?